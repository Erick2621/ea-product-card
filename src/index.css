/* ==== Variables y estilos base ==== */
:root {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  /* Usamos el de tu primer bloque */
  line-height: 1.5;
  font-weight: 400;

  color-scheme: light dark;
  color: white;
  /* Color base del primer bloque */
  background-color: #282c34;
  /* Fondo base del primer bloque */

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

/* ==== Elementos globales ==== */
html,
body {
  margin: 0;
  min-width: 320px;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  /* apila en columna */
  align-items: flex-start;
  /* alinea a la izquierda */
}

a {
  font-weight: 500;
  color: #646cff;
  text-decoration: inherit;
}

a:hover {
  color: #535bf2;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

h1 {
  font-size: 3.2em;
  line-height: 1.1;
}

button {
  border-radius: 8px;
  border: 1px solid transparent;
  /* padding: 0.6em 1.2em; */
  font-size: 1em;
  font-weight: 500;
  font-family: inherit;
  background-color: #1a1a1a;
  cursor: pointer;
  transition: border-color 0.25s;
}

button:hover {
  border-color: #646cff;
}

button:focus,
button:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}

/* ==== Layout principal ==== */
.main-layout {
  display: flex;
  flex-direction: row;
}

/* ==== Barra de navegaci√≥n ==== */
nav {
  background-color: #363a45;
  height: 100vh;
  margin-right: 15px;
  overflow-y: scroll;
  text-align: center;
  width: 250px;
}

nav ul {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-left: 0;
}

nav li {
  list-style: none;
  margin-bottom: 10px;
}

nav li a {
  color: white;
  font-size: 1.2rem;
  text-decoration: none;
}

.nav-active {
  color: grey;
  transition: all .3s ease-out;
}

/* ==== Modo claro ==== */
@media (prefers-color-scheme: light) {
  :root {
    color: #213547;
    background-color: #ffffff;
  }

  a:hover {
    color: #747bff;
  }

  button {
    background-color: #f9f9f9;
  }
}