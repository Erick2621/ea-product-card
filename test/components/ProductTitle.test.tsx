import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1, product2 } from '../data/products';
import { domToJSON } from '../functions/domToJSON';


describe('ProductTitle', () => {
    test('debe de mostrar el componente correctamente con el titulo personalizado', () => {

        const { container } = render(<ProductTitle title="Custom Product" className='custom-class' />);

        // Convertir el primer nodo hijo (el span) a JSON
        const json = domToJSON(container.firstChild as HTMLElement);
        console.log(json);

        expect(json).toMatchSnapshot();
    });

    test('debe de mostrar el componente  con el nombre del producto', () => {

        const { container } = render(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );

        // Convertir el primer nodo hijo (el span) a JSON
        const json = domToJSON(container.firstChild as HTMLElement);
        console.log(json);

        expect(json).toMatchSnapshot();
    });
});


