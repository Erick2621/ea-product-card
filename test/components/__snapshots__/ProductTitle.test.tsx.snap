// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductTitle > debe de mostrar el componente  con el nombre del producto 1`] = `
{
  "props": {
    "children": [
      {
        "props": {
          "children": [
            "Coffee Mug - Sin Imagen",
          ],
          "className": "_productDescription_3e9733 undefined",
          "style": undefined,
        },
        "type": "span",
      },
    ],
    "className": "_productCard_3e9733 undefined",
    "style": undefined,
  },
  "type": "div",
}
`;

exports[`ProductTitle > debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
{
  "props": {
    "children": [
      "Custom Product",
    ],
    "className": "_productDescription_3e9733 custom-class",
    "style": undefined,
  },
  "type": "span",
}
`;
